tasks/src/main/scala/dagr/tasks/gatk/GatkTask.scala

```diff
@@ -23,18 +23,43 @@
  */
 package dagr.tasks.gatk
 
-import java.nio.file.Path
+import java.nio.file.{Files, Path}
+import java.util.jar.JarInputStream
 
 import dagr.core.config.Configuration
 import dagr.core.execsystem.{Cores, Memory}
 import dagr.core.tasksystem.{FixedResources, ProcessTask}
-import dagr.tasks.{DagrDef, JarTask}
-import DagrDef.{PathToFasta, PathToIntervals}
+import dagr.tasks.JarTask
+import com.fulcrumgenomics.commons.CommonsDef._
 
 import scala.collection.mutable.ListBuffer
 
 object GatkTask {
   val GatkJarPathConfigKey = "gatk.jar"
+
+  /** Attempts to determine the major version from the GATK Jar. */
+  def majorVersion(jar: FilePath): Int = {
+    val in    = new JarInputStream(Files.newInputStream(jar))
+    val attrs = in.getManifest.getMainAttributes
+    in.close()
+
+    if (attrs.containsKey("Implementation-Version")) {
+      attrs.getValue("Implementation-Version").takeWhile(_ != '.').toInt
+    }
+    else {
+      attrs.getValue("Main-Class") match {
+        case "org.broadinstitute.sting.gatk.CommandLineGATK"  => 1
+        case "org.broadinstitute.gatk.engine.CommandLineGATK" => 3
+        case x => throw new IllegalArgumentException(s"Couldn't determind GATK version from jar $jar")
+      }
+    }
+  }
```

Comment:
This might be nice to put in some form in `JarTask`.

Reply:
I think the next time we do something like this we should do that.  In an ideal world the JAR would always have `Implementation-Version` in the manifest, but it would appear that GATK versions < 4 didn't do that, so then you have to start playing tricks to ID the version.

I think also, longer term we should probably have something like `case class MultiPartVersion(major: Int, minor: Int, patch: Option[Int], wtf: Option[Int])` and use that so that you can to work with minor versions etc.  But since I could only get major for < 4, I didn't think it was worth the time to do that yet.